Source/Xamarin/Prism.DryIoc.Forms/PrismApplication.cs

```diff
@@ -128,7 +128,7 @@ protected override void ConfigureViewModelLocator()
         /// <param name="view"><see cref="Page" /> navigated to</param>
         /// <param name="type"><see cref="Type" /> to resolve</param>
         /// <param name="navigationService">Overriding instance of <see cref="INavigationService" /></param>
-        protected virtual void ResolveTypeForPage(Page view, Type type, INavigationService navigationService)
+        protected internal virtual void ResolveTypeForPage(Page view, Type type, INavigationService navigationService)
```

Comment:
What if the test didn't involve testing the method directly, but rather we just test the result: creating a Page, then get the VM from the binding context (which has a public navigation service property, grab the nav service, and cast it as IPageAware, and then make the check.  Instead of trying to so tricks with methods and assembly attributes.